url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/AbstractPosition.java

```diff
@@ -0,0 +1,856 @@
+      {
+         List<Movement> possibleMoves = this.getMovements();
+         
+         if(possibleMoves.contains(mov))
+            return true;
+      }
+      
+      return false;
+   }
+   
+   public boolean isValidPosition()
+   {
+      //Si el jugador que no tiene el turno se encuentra
+      //en jaque la posición no es valida
+      if(getTurn() == Colour.WHITE)
+      {
+         if(isInCheck(Colour.BLACK))
+            return false;
+      }
+      else
+      {
+         if(isInCheck(Colour.WHITE))
+            return false;
+      }
+      
+      //No puede encontrarse ningún peon en la primera ni
+      //en la octava fila
+      if(hasGenericPiece(GenericPiece.PAWN, Colour.WHITE,Rank._1))
+         return false;
+      if(hasGenericPiece(GenericPiece.PAWN,Colour.BLACK,Rank._1))
+         return false;
+      if(hasGenericPiece(GenericPiece.PAWN,Colour.WHITE,Rank._8))
+         return false;
+      if(hasGenericPiece(GenericPiece.PAWN,Colour.BLACK,Rank._8))
+         return false;
+      
+      //Tiene que haber un rey por bando
+      //System.out.println("=>" + getNumeroPiezas(GenericPiece.REY, Colour.WHITE));
+      //if(getNumeroPiezas(GenericPiece.REY, Colour.WHITE) != 1)
+      // return false;
+      //if(getNumeroPiezas(GenericPiece.REY,Colour.BLACK) != 1)
+       //      return false;
+      
+      
+      //TODO hay que controlar el nuemro de piezas posibles;
+      
+      //TODO hay que controlar la casilla al paso
+      
+      //Si no ocurre ninguna de las condiciones anteriores la posición
+      //es válida
+      return true;
+   }
+   
+   public String toFen()
+   {
+      String fen = "";
+      
+      for(int i = 7; i >=0; i--)
+      {
+         String rank = "";
+         
+         int emptySquares = 0;
+         for(int j=0; j<8; j++)
+         {   
+            Piece piece = this.getPieceInSquare(Square.fromRankAndFile(i, j));
+            if(piece == null)
+               emptySquares++;
+            else
+            {
+               if(emptySquares > 0)
+               {
+                  rank += Integer.toString(emptySquares);
+               }
+               rank += piece.toFenString();
+               emptySquares = 0;
+            }
+         }
+         if(emptySquares > 0)
+         {
+            rank += Integer.toString(emptySquares);
+         }
+         
+         fen += rank;
+         if(i>0)
+            fen+= "/";
+      }
+      
+      String turn = this.getTurn() == Colour.WHITE ? "w" : "b"; 
+      
+      fen += " " + turn;
+      
+      String castlingRights = "";
+      if(this.getCastlingShort(Colour.WHITE))
+         castlingRights += "K";
+      if(this.getCastlingLong(Colour.WHITE))
+         castlingRights += "Q";
+      if(this.getCastlingShort(Colour.BLACK))
+         castlingRights += "k";
+      if(this.getCastlingLong(Colour.BLACK))
+         castlingRights += "q";
+      
+      fen += castlingRights == "" ? " -" : " " + castlingRights;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.